x-pack/plugins/observability/server/services/slo/summary_cleanup_task.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger, SavedObjectsClient } from '@kbn/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { SortResults } from '@elastic/elasticsearch/lib/api/types';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../assets/constants';
+import { StoredSLO } from '../../domain/models';
+import { SO_SLO_TYPE } from '../../saved_objects';
+
+export const TASK_TYPE = 'SLO-DEFINITIONS-CLEANUP-TASK';
+
+function findMissingItems(soIds: string[], sloDefIds: string[]) {
+  const missingSloDefIds = sloDefIds.filter((item) => !soIds.includes(item));
+  const missingSOIds = soIds.filter((item) => !sloDefIds.includes(item));
+
+  return {
+    missingSloDefIds,
+    missingSOIds,
+  };
+}
+
+export class SloSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClient;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '1m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              const runAt = new Date().toISOString();
+
+              if (this.soClient && this.esClient) {
+                const finder = this.soClient.createPointInTimeFinder<StoredSLO>({
+                  type: SO_SLO_TYPE,
+                  perPage: 1000,
+                  fields: ['id'],
+                });
+                let searchAfterKey: SortResults | undefined;
+                const soIds: string[] = [];
+                let sloDefIdsToDelete: string[] = [];
+
+                for await (const response of finder.find()) {
+                  const soItems = response.saved_objects.map((so) => so.attributes.id);
+
+                  const { sloDefIds, searchAfter } = await this.fetchSloDefinitions(searchAfterKey);
```

Comment:
Just a naming thing, but SLO definitions usually refers to the Saved Object, while here you are fetching the Summary documents. I think it would be less confusing to have this function/variable named `fetchSloSummaries` or something along this line.